source/fundamentals/logging.txt

```diff
@@ -0,0 +1,273 @@
+=======
+Logging
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. _logging:
+
+Overview
+--------
+
+.. What is on this page?
+
+In this guide, we show you how to set up and configure a logger in the
+MongoDB Java driver.
+
+.. What do any new terms mean?
+
+.. Not applicable. We are treating definition of logger as understood.
+
+.. What can you expect to see on this page?
+
+In this guide you can learn how to do the following:
+
+- Set up a logger using the Simple Logging Facade For Java (SLF4J)
+- Configure the log level of your logger
+
+.. Who should read this? Where should I go if this isn't the page I was looking for?
+
+.. Not Applicable. No natural redirect location.
+
+Set Up a Logger
+---------------
+
+To set up a logger for your project, you need to add some dependencies.
+
+The MongoDB Java driver uses the Simple Logging Facade For Java (SLF4J).
+SLF4J allows you to specify your logging framework of choice at deployment time.
+For more information on SLF4J,
+`see the SLF4J documentation <http://www.slf4j.org/>`__.
+
+To set up a logger you must add the ``slf4j-api`` package, a logging framework, and a
+**binding** to your project's dependency list. A binding is a piece of code that
+connects the ``slf4j-api`` package with a logging framework. The two most
+popular logging frameworks used with ``slf4j-api`` are Log4j2 and Logback.
+
+Setting up a logger is optional. When you start your application the MongoDB
+Java driver looks for the ``slf4j-api`` package in your classpath. If the driver
+can't find ``slf4j-api``, the driver logs a single warning with
+``java.util.logging`` and disables all further logging.
```

Comment:
A few items here.

1. It is stated that a particular dependency must be added, and that it creates a binding. Then the term binding is defined.

* Is including the term binding worthwhile? It doesn't appear to be required for the task of setting up a logger?

2. It is stated that the user MUST add the `slf4j-api` package. 
 
* Is that correct? [This post](https://stackoverflow.com/a/59374626/4279756) suggests an alternative based on desired framework.

3. The two most popular logging frameworks... feels incomplete.

* Perhaps rephrasing this to "This guide uses the two most popular logging framework..."


Lastly, this all feels very informational and not directly related to the task of "Set(ting) Up a Logger".  I wonder if information here could be extracted into an overview section?

Reply:
Thank you for the notes. I will respond to your feedback in order.

1.  I believe including the term binding is worthwhile.  The ``slf4j-api`` requires a separate framework specific binding package that binds it to a logging framework. This means that users must install three packages to use Logback or Log4j2. 
2. I believe it is correct that the user must add the ``slf4j-api`` in order to enable logging. The [current driver documentation](https://mongodb.github.io/mongo-java-driver/4.2/driver/reference/logging/) of logging states that you must install the ``sl4j-api`` in order to enable logging. Here is [the section of the Driver Source Code](https://github.com/mongodb/mongo-java-driver/blob/20c160a8906e6d65dc89e7f55b5d85f79f79e114/bson/src/main/org/bson/diagnostics/Loggers.java#L49) that will output a no-op logger is `slf4j-api` is not installed. SLF4J allows you to plugin your desired framework (it is a facade), so it is true that the user can use their desired framework.
3. I agree your proposed phrasing improves the sentence. I didn't realize how arbitrary and incomplete my wording sounded. Going forward, I will try to identify these incomplete phrases. Fix:
4. I agree that an overview/information section could be beneficial. I saw this suggestion in some of your other comments as well. I will experiment with this and try to group foundational knowledge into its own section.